src/services/data/data.service.ts

```diff
@@ -52,15 +37,17 @@ export class DataService implements OnDestroy {
           },
           name: {
             type: 'select',
-            defaultValue: name,
-            options: Object.keys(datasets),
+            defaultValue: defaultDatasetName,
+            options: datasetNames,
           },
           ...datasets[name].configMeta,
         };
 
         this.preferenceService.dataset$.next({
           ...createDefault(meta),
+          name,
```

Comment:
Why does there need to be 2 properties with the same value?

Reply:
There are two subscription in `DataService`: One for updating meta and one for creating data when preference changes. When user changes the `name` in preference, the `_meta_name` does not change, so the subscription above will not be triggered to create new data. After `_meta_name` is changed, which means the meta in preference is update to date, that subscription will then be triggered.

https://github.com/googleinterns/guide-doge/pull/67/files/09a2e79376e508f458b5aa82e4446a43bd5a1c03#diff-04c7fa6e46d3da32b3b6976c27dd53edR19